internal/gcp/secret-manager.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/minio/kes"
+	"google.golang.org/api/option"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+)
+
+type SecretManager struct {
+	// The GCP project id
+	// Obviously, you need to create GCP project
+	ProjectID string
+
+	// The GCP service account key json file
+	// Follow this documentation to create service account & get the key file
+	// https://cloud.google.com/secret-manager/docs/reference/libraries#setting_up_authentication
+	CredentialsFile string
+
+	// ErrorLog specifies an optional logger for errors
+	// when files cannot be opened, deleted or contain
+	// invalid content.
+	// If nil, logging is done via the log package's
+	// standard logger.
+	ErrorLog *log.Logger
+
+	client *secretmanager.Client
+}
+
+var (
+	errCreateKey = kes.NewError(http.StatusBadGateway, "bad gateway: failed to create key")
+	errGetKey    = kes.NewError(http.StatusBadGateway, "bad gateway: failed to access key")
+	errDeleteKey = kes.NewError(http.StatusBadGateway, "bad gateway: failed to delete key")
+)
+
+// Create stores the given key-value pair at GCP secret manager
+// if and only if it doesn't exists. If such an entry already exists
+// it returns kes.ErrKeyExists.
+//
+// GCP secret manager first need to create the secret itself
+// and then add a secret version which is where the actual key-value pair stored
+// The secret will be stored in encrypted form using AES-256
+func (s *SecretManager) Create(key, value string) error {
+	if s.client == nil {
+		s.logf("gcp: no connection to GCP secret manager: '%s'", s.ProjectID)
+		return errCreateKey
+	}
+
+	existingSecret, _ := s.client.GetSecret(
+		context.Background(),
+		&secretmanagerpb.GetSecretRequest{
+			Name: s.getSecretName(s.ProjectID, key),
+		},
+	)
+
+	if existingSecret != nil {
+		return kes.ErrKeyExists
+	}
+
+	secret, err := s.client.CreateSecret(
+		context.Background(),
+		&secretmanagerpb.CreateSecretRequest{
+			Parent:   fmt.Sprintf("projects/%s", s.ProjectID),
+			SecretId: key,
+			Secret: &secretmanagerpb.Secret{
+				Replication: &secretmanagerpb.Replication{
+					Replication: &secretmanagerpb.Replication_Automatic_{
+						Automatic: &secretmanagerpb.Replication_Automatic{},
+					},
+				},
+			},
+		},
+	)
+
+	if err != nil {
+		s.logf("gcp: failed to create '%s': %v", key, err)
+		return errCreateKey
+	}
+
+	_, err = s.client.AddSecretVersion(
+		context.Background(),
+		&secretmanagerpb.AddSecretVersionRequest{
+			Parent: secret.Name,
+			Payload: &secretmanagerpb.SecretPayload{
+				Data: []byte(value),
+			},
+		},
+	)
+
+	if err != nil {
+		s.logf("gcp: failed to create '%s': %v", key, err)
+		return errCreateKey
+	}
+
+	return nil
+}
+
+// Get returns the value associated with the given key.
+func (s *SecretManager) Get(key string) (string, error) {
+	if s.client == nil {
+		s.logf("gcp: no connection to GCP secret manager: '%s'", s.ProjectID)
+		return "", errGetKey
+	}
+
+	result, err := s.client.AccessSecretVersion(
```

Comment:
What happens if the secret does not exist. Is there no way to determine whether this is the case and return `kes.ErrKeyNotFound`